specs/semantics/hs/src/Control/State/Transition/Trace.hs

```diff
@@ -65,30 +67,144 @@ deriving instance
 mkTrace :: Environment s -> State s -> [(State s, Signal s)] -> Trace s
 mkTrace = Trace
 
+-- $setup
+-- |
+-- >>> :set -XTypeFamilies
+-- >>> import Control.State.Transition (initialRules, transitionRules)
+-- >>> :{
+-- data DUMMY
+-- instance STS DUMMY where
+--   type Environment DUMMY = Bool
+--   type State DUMMY = Int
+--   type Signal DUMMY = String
+--   data PredicateFailure DUMMY = CeciNEstPasUnePredicateFailure deriving (Eq, Show)
+--   initialRules = []
+--   transitionRules = []
+-- :}
```

Comment:
is this from Dante? This is nice, but can we guarantee automatic update in case of code changes ?

Reply:
This is `doctest`, so if we break the example we'll break the tests as well :)

Reply:
Oh, and it isn't from Dante. At first I thought you were referring to Alighieri, and that I was doing code changes worth of one of the 9 circles :P